cachecontrol/caches/uwsgi_cache.py

```diff
@@ -0,0 +1,40 @@
+try:
+    import uwsgi
+except ImportError:
+    # Not running in uwsgi
+    uwsgi = None
+
+from cachecontrol.cache import BaseCache
+from datetime import datetime
+
+class UWSGICacheError(Exception):
+    pass
+
+class UWSGICache(BaseCache):
+    def __init__(self, name=None):
+        if not uwsgi:
+            raise UWSGICacheError("uwsgi module is not available")
```

Comment:
To clarify on my previous comment in cache.py regarding the `clear()`, I think you could adjust this to do something like:

```
try:
    import uwsgi
    # Add your existing code
except ImportError:
   # Log a warning saying you're using a dict cache
   class UWSGICache(DictCache):
        def clear():
            pass
```
This way you're avoiding the exception when creating the object.

The reason you want to avoid errors from __init__ methods is that it makes it much harder to use the class. Every instantiation needs to be wrapped in a try/except block, which will change the scope of the variable making it hard to use. For example, often times when testing something, I'll instantiate some object purposefully skipping a critical attribute. The reason being is adding the extra data adds confusing noise, when my goal is to mock a specific method or use a method that doesn't require the attribute. 

I hope that is at least partially helpful describing why raising this error here is problematic.

Reply:
I don't really like the idea of the uwsgi cache still "working" when uwsgi is not present, and a simple warning might go unheeded in a production environment. But I get your point and it's still a better compromise than raising an exception in `__init__`, which I wasn't really comfortable doing anyway. So I'll make the change. Thanks !